packages/slate-react/src/components/editor.js

```diff
@@ -228,17 +163,59 @@ class Editor extends React.Component {
    */
 
   get schema() {
-    return this.state.schema
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.getSchemaWithMemoization(plugins)
   }
 
   get stack() {
-    return this.state.stack
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.getStackWithMemoization(plugins)
   }
 
   get value() {
-    return this.state.value
+    if (this.tmp.value === this.props.value) return this.tmp.value
+    return this.processValueOnChange(this.props.value, this.stack)
   }
 
+  /**
+   * Get stack by plugins, it is a memoized function and therefore do not need
+   * to re-evaluate for the same plugins
+   *
+   * @param {Array<Object>} plugins
+   * @return {Stack}
+   */
+
+  getStackWithMemoization = memoize(plugins => {
```

Comment:
Can you name these `resolveStack/resolveSchema` to match the existing `resolvePlugins`?

Why not have these call `resolvePlugins` internally themselves instead of needing to take in plugins? (Might be a good reason for it, I'm just quickly reading.)

Reply:
Fixed~

Reply:
Because we need a argument to make the function working with right cache.  Therefore we would like to pass the plugins.  If we do not have plugins as argument, then it will always return the same value regardless of plugins change